[API] integrated rick&morty api

fedorovychh opened the pull request:
1. added all needed dependencies into pom file
2. added configuration for mapper
3. added custom ApplicationRunner
4. added service
5. added controller
6. added repository
7. added swagger
8. added entity
9. added dtos
10. added mapper
![screenshoot-swagger](https://github.com/mate-academy/jv-rick-and-morty/assets/137708693/49ee2b37-7f66-4b60-85de-15c3c7271659)

fedorovychh committed:
third-party api integrated

fixed string to digit

fixed checkstyle

realized character controller

realized getRandom logic

refactored apprunner

refactored apprunner

realized adding characters to db

added lombok

added Characted entity

added dependencies and configuration

d1sam commented on src/main/java/mate/academy/rickandmorty/service/impl/ApplicationRunnerServiceImpl.java:
Please don't violate SRP. You should have 2 different classes: 
1. for client to http third-party
2. for initialization of db

d1sam replied:
Also it's better to use @PostConstruct to initialize db

d1sam commented on src/main/java/mate/academy/rickandmorty/service/impl/CharacterServiceImpl.java:
```diff
@@ -0,0 +1,35 @@
+@Service
+@RequiredArgsConstructor
+public class CharacterServiceImpl implements CharacterService {
+    private static final int MAX_CHARACTERS_LIMIT = 827;
```
let's get this value from db, don't hard-code

d1sam reviewed (changes requested):
Good job, but there are some changes that should be done!

fedorovychh committed:
improved according to SRP

changed name method

separated DB transaction out of AppRunnerService

changed way of getting random character

d1sam reviewed (approved):
well done!
